shiksha-platform/frontend-modulefederation
# Refactoring CalendarBar in Attendance Module

## Description
As per suggestions, following a step by step approach for the migration and refactoring for the attendance module.

### Changes
Made the following changes to the attendance module:

* added typescript support
* migrated the calendar bar component to typescript and also refactored it in the process
* also fixed the imports of the calendar bar component in other modules
* for the calendar bar, instead of having daywise, weekwise, and monthwise components separately made a single component "TimeBar" that accepts view as a parameter to render them as per requirement.


## How to test
1. Use the command 
```lerna run start --parallel --scope='{attendance,core,@shiksha/common-lib,students}'``` 
to fire up just the attendance module and its related dependencies.
2. Now the attendance module should behave completely normal since no change in anything else.


### Modified Calendar Bar
![feature](https://user-images.githubusercontent.com/82583557/181235858-1e1f9003-dc02-4099-9570-e4b63d3f5aba.gif)
